src/main/java/mate/academy/internetshop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,115 @@
+package mate.academy.internetshop.dao.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.internetshop.dao.ProductDao;
+import mate.academy.internetshop.exceptions.DataProcessingException;
+import mate.academy.internetshop.lib.Dao;
+import mate.academy.internetshop.model.Product;
+import mate.academy.internetshop.util.ConnectionUtil;
+import org.apache.log4j.Logger;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+    private static final Logger LOGGER = Logger.getLogger(ProductDaoJdbcImpl.class);
+
+    @Override
+    public Product create(Product product) {
+        String query = "INSERT INTO products (name, price) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setString(1, product.getName());
+            preparedStatement.setBigDecimal(2, product.getPrice());
+            if (preparedStatement.executeUpdate() > 0) {
+                ResultSet resultSet = preparedStatement.getGeneratedKeys();
+                resultSet.next();
+                product.setId(resultSet.getLong(1));
+            } else {
+                LOGGER.warn("Can't create product");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create product " + e);
+        }
+        return product;
+    }
+
+    @Override
+    public Product update(Product product) {
+        String query = "UPDATE products SET name = ?, price = ? WHERE `product.id`= ?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setString(1, product.getName());
+            preparedStatement.setBigDecimal(2, product.getPrice());
+            preparedStatement.setLong(3, product.getId());
+            if (preparedStatement.executeUpdate() == 0) {
+                LOGGER.warn("Can't update product with id: " + product.getId());
+            }
```

Comment:
same

Reply:
Here we can try to update the product which doesn't exist in the database, for example, if it was removed. Shouldn't we log such situations?